sampling/thapi_sampling.h

```diff
@@ -0,0 +1,5 @@
+#include <time.h>
+
+extern int thapi_sampling_init();
+
+extern void thapi_register_sampling(void (*pfn)(void), struct timespec *interval);
```

Comment:
```suggestion
/*
 * TODO: Updtae API to have it take a state:
 * rename and change siganture to:
 * extern int thapi_register_sampling_callback(struct timespec *interval, int (*pfn)(void*), void* user_data);
 * so Callbakcs can take an additional state and return an status code (that could be used to stop
 * unregister the callabck)
 */
extern void thapi_register_sampling(void (*pfn)(void), struct timespec *interval);
```
Adding this here for later.